Several improvements to the dashboard layout

l0ll098 opened the pull request:
Hi there!
I've made some changes to the dashboard to improve it a little bit.

In particular, this pull request contains changes as briefly described here:

 - Several improvements to the layout:
    - improved positioning of the `scope` select
    - dashboard will now fill the entire screen (both vertically and horizontally)
    - dashboard layout will now be more responsive (still not perfect on smartphones but at least now it can be used with tablets)
    - decluttered template code removing redundand classes and ids, useless styles, ...
    - sidenav and content placement are now automatically handled by the browser (no longer need to update the `margin-left` propery of the content section each time sidenav is toggled)
    - added a missing label in `about` button
 - Preferred theme is now persistent across page reloads.
    Each time a user changes theme color, his/her selection is stored in `localStorage`. When (s)he returns to the dashboard, we will use previously saved theme.
 - Showing Dapr Dashboard information in "About" popup. Those includes: 
    - Dapr version (e.g.: `edge`)
    - Dapr dashboard version (e.g.: `0.2.0`)
    - Dapr dashboard hash (e.g.: `g279393c`)
 - The "About" popup has now two buttons: "close" and "copy info". The latter one, as the name suggests, allows you to quickly copy information about your Dapr setup (currently dapr version, dapr dashboard version and hash)

Dapr dashboard details (version and hash) are generated using a custom NodeJS script stored in the `web/scripts` folder and automatically executed once npm packages are installed (through the `postinstall` hook).
This script will then get some info from Git, including the hash shown in the "About" popup. Dashboard version is instead being read from the `package.json` file located in the `web` folder. At this point a new file called `web/src/environments/version.ts` will be written and it will contain those details. Since it has been added to the `.gitignore`, it won't get accidentally committed.

The latter point should also address issue #148.

I'm also attaching a screenshot of the dashboard itself that, as you can see, is not changed very much since the majority of my changes are under the hood things.

![dashboard](https://user-images.githubusercontent.com/13736036/126039094-da7f1d0b-378e-44af-aa4b-cc9f79702648.png)

In case you have any questions/suggestions/doubts (or whatever), feel free to ping me :)

And, finally, have a nice day

l0ll098 commented:
Just made some more changes to the codebase in order to improve the part that shows YAML configuration files.
Actually, before my last commit, if you were to use the dark mode, those snippets would have been showed with the same theme as if you were using the light mode. Clearly this resulted in a poor experience, since it was quite hard to read the output.

So here's a more complete list of things I did in this last commit:
 - Removed redundant code used to render yaml configuration files by adding a new component (called `EditorComponent` and stored in a `SharedModule`) that will automatically handle all the heavy lifting
 - `EditorComponent` shows a Monaco Editor style element with your code. Currently it exposes only a `model` property that can be used to pass the content to show. It is also ready to reflect and propagate changes to external components but, since the monaco editor is set to be readonly, you won't be able to see those changes anywhere
 - Changed every section of the dashboard that was using the old approach in order to use this new component


I'm also going to attach a couple of screenshots so that you can more easily see the result :)

With dark theme on:
![dark_theme](https://user-images.githubusercontent.com/13736036/126062726-eb7d176b-1e50-4ae0-92d2-9d1b2a653e9d.jpeg)

With light theme:
![light_theme](https://user-images.githubusercontent.com/13736036/126062727-db766079-5ff0-464f-a1b7-3e1c0c69a0c4.jpeg)

yaron2 commented:
/cc @artursouza

artursouza commented:
@l0ll098 Thanks a lot for this PR. This is really great work. Can you change the copy feature to work per field? This would be similar to how Azure show a copy icon next to a text:

![image](https://user-images.githubusercontent.com/130954/126853106-e74329d4-e099-4424-9e09-89982eed04b4.png)

l0ll098 commented:
Hi @artursouza,
thanks for your reply :)

With that being said: I've fixed as you requested. Each row has a button to copy just that single piece of information (so for example you will be able to copy just the dapr version, with no additional text).
I've also changed the label of the previous "copy" button (the one that copied everything showed in the popup) to "Copy all". It's behaviour is still the same. This could be used, for example, when opening a new issue on Github as it would probably help to narrow down the root cause of the reported issue.

yaron2 commented:
/cc @artursouza

yaron2 commented:
@l0ll098 thanks for these improvements, great to see.
